app/routes/_extras.tsx

```diff
@@ -1,15 +1,56 @@
+import React from "react";
 import { Outlet } from "react-router";
 import { DocSearchModal } from "~/ui/docsearch";
 import { Footer } from "~/ui/footer";
 import { Header } from "~/ui/header";
 
 export default function ExtrasLayout() {
+  return (
+    <Layout>
+      <Outlet />
+      <Footer />
+    </Layout>
+  );
+}
+
+// Force this layout HydrateFallback to render on initial load so we can
+// eliminate the footer and avoid needing to fill height with a skeleton
+export function clientLoader() {
+  return null;
+}
+clientLoader.hydrate = true;
```

Comment:
This might be a bug or a gap in RR - I thought we bubbled to the nearest `HydrateFallback` but it looks like it's 1:1 so just having the `clientLoader.hydrate` on the child route is not enough to get this route `HydrateFallback` to show.  I wanted to do it at this level so we didn't have the footer collapse up on us when there was no content and cause a flicker.  By doing it here I can just hide the footer until the content is ready.

Reply:
hmmm yeah, feels like it should work the way you expected. Either way, this approach makes sense